gtsam/discrete/TableFactor.h

```diff
@@ -178,6 +178,10 @@ class GTSAM_EXPORT TableFactor : public DiscreteFactor {
   /// multiply with DecisionTreeFactor
   DecisionTreeFactor operator*(const DecisionTreeFactor& f) const override;
 
+  /// Multiply factors, DiscreteFactor::shared_ptr edition
```

Comment:
Same. Provide meaningful comment about return type and complexity implicafions of passing one version vs other